gb-tracy/scripts/articleView.js

```diff
@@ -0,0 +1,119 @@
+    if ($(this).val()) {
+      // DONE: If the <select> menu was changed to an option that has a value, we first need to hide all the articles, and then show just the ones that match for the author that was selected.
+      // Use an "attribute selector" to find those articles, and fade them in for the reader.
+      $('article').hide();
+      $(`article[data-author="${$(this).val()}"]`).fadeIn(1000);
+    } else {
+      // DONE: If the <select> menu was changed to an option that is blank, we should first show all the articles, except the one article we are using as a template.
+      $('article').fadeIn(500);
+      $('article[class="template"]').hide();
+    }
+    $('#category-filter').val('');
+    $('.article-body *:nth-of-type(n+2)').hide();
+    $('.article-body img').hide();
+  });
+};
+
+articleView.handleCategoryFilter = function() {
+  // DONE: Just like we do for #author-filter above, we should handle change events on the #category-filter element.
+  // When an option with a value is selected, hide all the articles, then reveal the matches.
+  // When the blank (default) option is selected, show all the articles, except for the template.
+  // Be sure to reset the #author-filter while you are at it!
+  $('#category-filter').on('change', function() {
+    if ($(this).val()) {
+      $('article').hide();
+      $(`article[data-category="${$(this).val()}"]`).fadeIn(1000);
+    } else {
+      $('article').fadeIn(500);
+      $('article[class="template"]').hide();
+    }
+    $('#author-filter').val('');
+  });
+};
+
+articleView.handleMainNav = function() {
+  // DONE: Add an event handler to .main-nav elements that will power the Tabs feature.
+  // Clicking any .tab element should hide all the .tab-content sections, and then reveal the single .tab-content section that is associated with the clicked .tab element.
+  // So: You need to dynamically build a selector string with the correct ID, based on the data available to you on the .tab element that was clicked.
+  $('.main-nav').on('click', 'li', function() {
+    let clickNav = $(this).data('content');
+    if (clickNav) {
+      $('.tab-content').hide();
+      $(`#${clickNav}`).fadeIn(1000);
+    }
+  })
+
+  // REVIEW: Now trigger a click on the first .tab element, to set up the page.
+  $('.main-nav .tab:first').click();
+};
+
+articleView.setTeasers = function() {
+  // REVIEW: Hide elements beyond the first 2 in any article body.
+  $('.article-body *:nth-of-type(n+2)').hide();
+  $('.article-body img').hide();
+
+  // DONE: Add an event handler to reveal all the hidden elements, when the .read-on link is clicked. You can go ahead and hide the "Read On" link once it has been clicked. Be sure to prevent the default link-click action!
+  // Ideally, we'd attach this as just one event handler on the #articles section, and let it process (in other words... delegate) any .read-on clicks that happen within child nodes.
+  $('#articles').on('click', 'a', function(e) {
```

Comment:
Watch out! Since you do this to all `a` elements, the links to the author's website don't work anymore 😭